rundeck-data-model/src/main/java/org/rundeck/app/data/model/v1/storage/RundeckStorage.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Rundeck, Inc. (http://rundeck.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.rundeck.app.data.model.v1.storage;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.rundeck.app.data.utils.Utils;
+
+import java.io.Serializable;
+import java.util.*;
+
+public interface RundeckStorage {
+    Serializable getId();
+    String getNamespace();
+    String getDir();
+    String getName();
```

Comment:
i think the interface should return a single Path object, instead of dir and name, that is the more direct translation of the datatype